### PR TITLE
#496 Read model for StaffUser

### PR DESCRIPTION
**Pull request by woksin**
Updated the Read-model for StaffUsers with a polymorphic MongoDB collection by using a discriminator based on the StaffUser roles.

Left to do:
  Update endpoints
  Handle adding phonenumbers and assigned national societies

### CODE REVIEW

**Comment by Velcrow81** [comment]
@smithmx and @Davlind : You have a better overview of User managment, so it would be nice if you could have a look too since there are a loot of changes

File: Source/UserManagement/Domain/DataCollector/DataCollector.cs [hunk 1]
```diff
@@ -118,20 +119,20 @@ private void RemovePhoneNumbers(IEnumerable<string> numbers)
...
-        {
-            _numbers.Add(@event.PhoneNumber);
-        }
+        //private void On(PhoneNumberAddedToDataCollector @event)
+        //{
+        //    _numbers.Add(@event.PhoneNumber);
+        //}
 
-        private void On(PhoneNumberRemovedFromDataCollector @event)
-        {
```
[the remark this answers is not in the document]
Comment (woksin) [reply Source/UserManagement/Domain/DataCollector/DataCollector.cs]: My bad, leftover from Code eve

**Comment by woksin** [comment]
It should be merge-ready now.

I've done testing on the generic collection in a controller method that I've removed.
I think that most of the wanted functionality is in place. 

@Velcrow81 Something like this would be nice in VolunteerReporting aswell for the *CaseReport* collections

**Comment by Velcrow81** [comment]
Seems to be a build error because of NuGet in admin project. It might be the same I have. 

Create an issue to fix this for case reports 😊

**Comment by karolikl** [comment]
See #534 for update on the missing nuget.

**Pull request merged**